mqtt/.dockerignore

```diff
@@ -3,6 +3,7 @@
 .git
 .gitignore
 .vscode
+state
```

Comment:
what app generates this folder?

Reply:
mqttd generates this folder to persist state.

Reply:
should we add it to a `.gitignore` as well?

Reply:
it is already there - https://github.com/Azure/iotedge/blob/master/mqtt/.gitignore